src/server/pfs/server/driver.go

```diff
@@ -656,16 +653,23 @@ func (d *driver) createProjectInTransaction(txnCtx *txncontext.TransactionContex
 }
 
 func (d *driver) inspectProject(ctx context.Context, project *pfs.Project) (*pfs.ProjectInfo, error) {
-	pi := &pfs.ProjectInfo{}
-	if err := d.projects.ReadOnly(ctx).Get(pfsdb.ProjectKey(project), pi); err != nil {
-		return nil, errors.EnsureStack(err)
+	var pi *pfs.ProjectInfo
+	var err error
+	if err = dbutil.WithTx(ctx, d.env.DB, func(ctx context.Context, tx *pachsql.Tx) error {
```

Comment:
Also, something I worry about in general is what happens when something transactional wants to call inspectProject.  I don't think it ever happens, so there is nothing to change here today, but it's why we see `xxxInTransaction` methods throughout the codebase.

Especially below when we AuthServer.GetPermissions(), ideally that would be the same view of the database as we had when retrieving the project.  But, Auth will probably eventually be its own service and not be able to be transactional with projects, so I'm not super bothered.  Something to keep in mind though, as the refactor proceeds.